### PR TITLE
Fix offset for iOS 13

### PR DESCRIPTION
**Pull request by ryannair05**

### CODE REVIEW

File: Tweak.xm [hunk 1]
```diff
@@ -108,6 +90,25 @@ static CGFloat offset = 0;
 %end	// %hook SBUICAPackageView
 
 %hook SBFLockScreenDateView
+-(id)initWithFrame:(CGRect)arg1 {
+
+    if (%c(JPWeatherManager) != nil) {
+		%orig;
+		return;
+    }
```
Comment (SniperGER) [inline Tweak.xm]: **Required:** This part needs to `return %orig`.

File: Tweak.xm [hunk 2]
```diff
@@ -108,6 +90,25 @@ static CGFloat offset = 0;
 %end	// %hook SBUICAPackageView
 
 %hook SBFLockScreenDateView
+-(id)initWithFrame:(CGRect)arg1 {
```
Comment (SniperGER) [inline Tweak.xm]: _Note:_ Keep the formatting in line with the rest of the code. Not necessary, but it looks nicer.
`- (id)initWithFrame:(CGRect)arg1 {` (note the space between `-` and `(id)`)

File: Tweak.xm [hunk 3]
```diff
@@ -209,6 +210,15 @@ static CGFloat offset = 0;
...
 %end	// %hook BSUICAPackageView
+
+%hook CSCombinedListViewController
+- (UIEdgeInsets)_listViewDefaultContentInsets {
+    UIEdgeInsets orig = %orig;
+
+    orig.top += offset;
+    return orig;
+}
+%end	// %hook CSCombinedListViewController
```
Comment (SniperGER) [inline Tweak.xm]: _Note:_ `%hook`s inside `%group`s are generally commented with three slashes in my code, as it keeps things a little more structured in my opinion. Again, this isn't necessary whatsoever, but I think it's nicer to keep a consistent code style.

**Review by SniperGER — CHANGES_REQUESTED** [review]
Confirmed to be working on iOS 12 (iPhone 5s) and iOS 13 (iPad Pro 10.5"). However, it doesn't seem to make any difference.

Please note that moving the offset "calculation" to `- [SBFLockScreenDateView initWithFrame:]` requires every code path to have a return value, which isn't given by your check for `JPWeatherManager`. Probably a Copy & Paste error?

**Comment by ryannair05** [comment]
Yeah, sorry about that copy and paste area. If you try to play media or get a notification these changes fix them from overlapping other components of the lockscreen. I also changed the MSHookFunction to only run on iOS 12 as from my testing it is not needed on iOS 13

**Comment by SniperGER** [comment]
Thanks for the update! I'll have a look at it later today.

**Comment by SniperGER** [comment]
Everything seems to be working on my testing devices so far, so I'm going to merge this. Now we just need a solution to the padlock overlapping the passcode UI on smaller devices, such as the iPhone 5s. Maybe there's a way to visually hide it when it transitions to the passcode buttons (as an optional setting, of course).

**Pull request merged**